src/service-override/extensionGallery.ts

```diff
@@ -30,14 +30,45 @@ import 'vs/workbench/contrib/logs/common/logs.contribution'
 import { IRemoteUserDataProfilesService, RemoteUserDataProfilesService } from 'vs/workbench/services/userDataProfile/common/remoteUserDataProfiles'
 import { ExtensionEnablementService } from 'vs/workbench/services/extensionManagement/browser/extensionEnablementService'
 import 'vs/workbench/services/extensionManagement/browser/extensionBisect'
+import { IInstantiationService, ILabelService, IRemoteAgentService } from '../services'
 
-export default function getServiceOverride (): IEditorOverrideServices {
+class ExtensionManagementServerServiceOverride extends ExtensionManagementServerService {
+  constructor (
+    private isWebOnly: boolean,
+    @IRemoteAgentService readonly remoteAgentService: IRemoteAgentService,
+    @ILabelService readonly labelService: ILabelService,
+    @IInstantiationService readonly instantiationService: IInstantiationService
+  ) {
+    super(remoteAgentService, labelService, instantiationService)
+
+    if (this.isWebOnly) {
+      /**
+      * If `isWebOnly` is set to true, we explicitly set the remote extension management server to `null`, even if
+      * we're connected to a remote server.
+      */
+      // Cannot override read-only property, but this is the only way we can override it to be null,
+      // overriding the field doesn't work and setting a getter is not allowed.
+      // @ts-ignore
+      this.remoteExtensionManagementServer = null
+    }
+  }
+}
+
+export interface ExtensionGalleryOptions {
+  /**
+   * Whether we should only allow for web extensions to be installed, this is generally
+   * true if there is no server part.
+   */
+  webOnly: boolean
+}
+
+export default function getServiceOverride (options: ExtensionGalleryOptions = { webOnly: false }): IEditorOverrideServices {
```

Comment:
Can we make the parameter optional with a default value compliant to the VSCode behavior?

Reply:
It's optional as it has an initializer:

<img width="1057" alt="image" src="https://github.com/CodinGame/monaco-vscode-api/assets/587016/ef7e78a1-0eb1-4833-8bea-64248d4f98bd">

Reply:
Ok I need to sleep more, my bad